R/transformAbundance.R

```diff
@@ -0,0 +1,400 @@
+#'
+#' @examples
+#' data(esophagus)
+#' x <- esophagus
+#'
+#' # By default, returns object with identity table
+#' x <- transformAbundance(x)
+#' assays(x)$identity
+#'
+#' # By specifying, it is possible to apoly different transformations, e.g. clr transform
+#' # Pseudocount can be added by giving the value TRUE. Then pseudocount is 1.
+#' x <- transformAbundance(x, transform="clr", pseudocount=TRUE)
+#' assays(x)$clr
+#'
+#' # Name of the stored table, can be specified. Also, the target of transform
+#' # can be specified with "abund_values". I addition to pseudocount=TRUE, it is
+#' # also possible to specify it by giving numeric value.
+#' x <- transformAbundance(x, transform="hellinger", name="test", pseudocount=5)
+#' assays(x)$test
+#' x <- transformAbundance(x, transform="Z", abund_values="test")
+#' assays(x)$Z
+#' # Z-transform can also be done for features
+#' x <- transformAbundance(x, transform="Z", pseudocount=TRUE, target="features")
+#' assays(x)$Z
+#'
+#' # With getTransformAbundance, it is possible to get only the transformed abundance
+#' # table without being stored in assays.
+#' getTransformAbundance(x, transform="log10", pseudocount=1)
+#'
+#'
+NULL
+
+#' @rdname transformAbundance
+#' @export
+setGeneric("transformAbundance", signature = c("x"),
+           function(x,
+                    abund_values = "counts",
+                    transform = "identity",
+                    name = transform,
+                    pseudocount = FALSE,
+                    scale = 1,
+                    target = "samples")
+               standardGeneric("transformAbundance"))
+
+
+#' @rdname transformAbundance
+#' @export
+setMethod("transformAbundance", signature = c(x = "SummarizedExperiment"),
+          function(x,
+                   abund_values = "counts",
+                   transform = "identity",
+                   name = transform,
+                   pseudocount = FALSE,
+                   scale = 1,
+                   target = "samples"){
+
+              # Input check
+              # Check abund_values
+              .check_abund_values(abund_values, x)
+
+              # Check transform
+              if(!(transform %in%
+                   c("relabundance", "Z", "log10", "log10p", "hellinger", "identity", "clr"))){
+                  stop("'transform' must be one of following methods 'relabundance',
+                       'Z', 'log10', 'log10p', 'hellinger', 'identity', 'clr'.",
+                       call. = FALSE)
+              }
+
+              # Check name
+              if(!.is_non_empty_string(name)){
+                  stop("'name' must be a non-empty single character value.",
+                       call. = FALSE)
+              }
+
+              # Check pseudocount
+              if(!(pseudocount==TRUE || pseudocount==FALSE || pseudocount>0)){
+                  stop("'pseudocount' must be boolean or positive numeric value.",
+                       call. = FALSE)
+              }
```

Comment:
see above

Reply:
--> "'pseudocount' must be FALSE or numeric value."